libpages/stathat.go

```diff
@@ -0,0 +1,141 @@
+// Copyright 2018 Keybase Inc. All rights reserved.
+// Use of this source code is governed by a BSD
+// license that can be found in the LICENSE file.
+
+package libpages
+
+import (
+	"strconv"
+	"strings"
+	"time"
+
+	stathat "github.com/stathat/go"
+	"go.uber.org/zap"
+)
+
+type stathatReporter struct {
+	logger   *zap.Logger
+	ezKey    string
+	reporter stathat.Reporter
+
+	statNameRequests      string
+	statNameAuthenticated string
+	statNameCloningShown  string
+	statNameInvalidConfig string
+	statPrefixProto       string
+	statPrefixStatus      string
+	statPrefixTlfType     string
+	statPrefixRootType    string
+
+	activityStats         ActivityStatsEnabler
+	statPrefixActiveHosts string
+	statPrefixActiveTlfs  string
+}
+
+func (s *stathatReporter) activityStatsReportLoop() {
+	if len(s.activityStats.Durations) == 0 || s.activityStats.Interval == 0 {
+		return
+	}
+
+	durations := make([]time.Duration, len(s.activityStats.Durations))
+	copy(durations, s.activityStats.Durations)
+	statNamesHosts := make([]string, 0, len(durations))
+	statNamesTlfs := make([]string, 0, len(durations))
+	for _, d := range durations {
+		statNamesHosts = append(statNamesHosts, s.statPrefixActiveHosts+"("+d.String()+")")
```

Comment:
The string for `time.Hour * 24 * 7` is "168h0m0s".  We might want to pass in a map with a more friendly label.